substrate/frame/assets/src/traits.rs

```diff
@@ -0,0 +1,18 @@
+pub mod sufficiency {
+	use sp_runtime::DispatchResult;
+
+	/// Trait for providing the sufficiency of an asset.
+	pub trait Inspect<AssetId> {
+		/// Returns whether an asset is sufficient or not.
+		fn is_sufficient(asset_id: AssetId) -> bool;
+	}
+
+	/// Trait for mutating the sufficiency of an asset
+	pub trait Mutate<AssetId> {
```

Comment:
I would try to come up with a different namings for new traits. otherwise it will create more confusions. also I see the possibility for these traits moving support module